### ISSUE TITLE
Cannot override serializeDate()

### ISSUE THREAD

**Issue by wgprojects** [laravel/framework#11185]
In [this](https://github.com/laravel/framework/commit/fd5e1a7026d88e27794f2db08b788944421275a3) commit, Taylor added serializeDate() at line 2968 of Model.php.
I attempted to override this in my model by copying into my BaseModel (the base for my models):
```PHP
protected function serializeDate(DateTime $date)
{
    return $date->format('insert my customized date format string here');
}
```

I get this error:
Uncaught exception 'ErrorException' with message 'Argument 1 passed to App\\BaseModel::serializeDate() **must be an instance of App\DateTime, instance of Carbon\Carbon given**, called in (snip)\vendor\laravel\framework\src\Illuminate\Database\Eloquent\Model.php on line 2480 and defined' in (snip)\app\BaseModel.php

Sticking a `var_dump($date);` in the Model.php serializeDate() shows the parameter $date really is a object(Carbon\Carbon). 

I have also tried changing the type (removing the DateTime in the parameter)
```PHP
protected function serializeDate($date)
...
```
but I get the error:
Declaration of App\BaseModel::serializeDate() should be compatible with Illuminate\Database\Eloquent\Model::serializeDate(DateTime $date)"

1. What is converting the $date to a Carbon instead of a DateTime?
    (I'm sure it's something I'm doing wrong, but what?)
2. How can I serialize my dates when I convert them to JSON (in a generic way), since this isn't working for me?
As a workaround, I'm creating `public function getCreatedAtAttribute` etc for each date stamp I want to convert.

P.s. I previously tried the $dateFormat property, but that made it try to write ISO8601 to my database, when I only want that format in my JSON. I'm not sure if this has changed - I'll look into it, but IMHO it doesn't have a bearing on the above not working.

**Comment by GrahamCampbell** [comment]
Looks like a missing import from us.

**Comment by GrahamCampbell** [comment]
Oh, wait, no. This is an error in your app.

**Comment by GrahamCampbell** [comment]
> protected function serializeDate(DateTime $date)

You MUST `use DateTime;`.

**Comment by wgprojects** [comment]
I only tried removing DateTime after it didn't work as shown in the first code example.My code is not (intentionally) converting to a Carbon type - how is the function in Model.php accepting a Carbon when it should be a DateTime? And why won't the same code work in my base model when I simply copy-paste from model.php?
I acknowledge that it works for most people, I'm just trying to track down this incongruity in what works in vendor code vs my base model - hoping for help, to figure out why the serialize function is being passed (apparently) the wrong type - I.e. what I did wrong.
ThanksWilliam

Sent from Yahoo Mail on Android 
 
  On Thu, Dec 3, 2015 at 5:38 PM, Graham Campbell<notifications@github.com> wrote:   

protected function serializeDate(DateTime $date)


You MUST use DateTime;.
